src/sentry/static/sentry/app/utils/parseLinkHeader.tsx

```diff
@@ -0,0 +1,25 @@
+export default function parseLinkHeader(
+  header: string | null
+): {[key: string]: {href: string; results: boolean | null; cursor: string}} {
+  if (header === null) {
+    return {};
+  }
+
+  const header_vals = header.split(',');
+  const links = {};
+
+  header_vals.forEach(val => {
+    const match = /<([^>]+)>; rel="([^"]+)"(?:; results="([^"]+)")?(?:; cursor="([^"]+)")?/g.exec(
+      val
+    );
+    const hasResults = match![3] === 'true' ? true : match![3] === 'false' ? false : null;
+
+    links[match![2]] = {
+      href: match![1],
```

Comment:
What do these `!` do?

Reply:
It's the non-null assertion operator.

https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#non-null-assertion-operator